app/src/org/commcare/CommCareApplication.java

```diff
@@ -275,9 +277,8 @@ protected void turnOnStrictMode() {
     }
 
     @Override
-    public void onConfigurationChanged(Configuration newConfig) {
+    public void onConfigurationChanged(@NonNull Configuration newConfig) {
         super.onConfigurationChanged(newConfig);
-        LocalePreferences.saveDeviceLocale(newConfig.locale);

```

Comment:
I think this line may still be necessary for other parts of the code

Reply:
Yes, that's correct. This line is already present in the onCreate method, which is why I removed it from here.

Reply:
Yes, LocalePreferences should update correctly after changing the language while the app is running.

I have already tested this scenario.